Add references to Nbshell in export templates, reports and scripts (#15651)
netbox-community/netbox: docs/administration/netbox-shell.md

# The NetBox Python Shell

> NetBox includes a Python management shell within which objects can be directly queried, created, modified, and deleted. To enter the shell, run the following command from a shell that has the netbox virtualenv activated:

Reviewer: ```suggestion
NetBox includes a Python management shell within which objects can be directly queried, created, modified, and deleted. To enter the shell, run the following command with the appropriate Python virtual environment activate:
```

>>> IPAddress.objects.filter(interface__device__site__region__slug="north-america")
```

> The `get()` method can be used in place of the `filter()` method if we expect this query to return only one object. This method will yield the actual object resulting from the query, instead of a QuerySet. If the query returns more than one result, it will raise an exception.

Reviewer: This is redundant to information provided in the [Django documentation](https://docs.djangoproject.com/en/5.0/ref/models/querysets/). Please reference it here instead of duplicating text that we'll need to then maintain.

  File "/srv/netbox/venv/lib/python3.10/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/srv/netbox/venv/lib/python3.10/site-packages/django/db/models/query.py", line 640, in get
    raise self.model.MultipleObjectsReturned(
dcim.models.devices.Device.MultipleObjectsReturned: get() returned more than one Device -- it returned more than 20!
>>>
```

Queries can all also be executed from a particular object instead of from the model itself. For instance, to get all circuits that are assigned to one site, it is easier to filter from the site itself, instead of using the "Circuit" model and building the query from there. This is particularly useful for configuration templates and export templates, since it allows to query other database objects that are related to the object that we're rendering the template for.

```
>>> site.circuit_terminations.all()
<RestrictedQuerySet [<CircuitTermination: 20899518: Termination A>, <CircuitTermination: DT00018356: Termination A>]>
>>>
```

The same methods (all, filter, exclude, get...) can be used in this kind of queries.

> 

Reviewer: This is out of scope for the NetBox documentation as it describes generic behavior of the [Django ORM](https://docs.djangoproject.com/en/stable/topics/db/queries/). Rather than duplicate Django documentation, we should refer users there to read further.

Author: Done. I reduced it to just one line about the get() method, and left the tip admonition pointing users to the Django documentation page

346
```

> Queries can all also be executed from a particular object instead of from the model itself. For instance, to get all circuits that are assigned to one site, it is easier to filter from the site itself, instead of using the "Circuit" model and building the query from there. This is particularly useful for configuration templates and export templates, since it allows to query other database objects that are related to the object that we're rendering the template for. The same methods (all, filter, exclude, get...) can be used in this kind of queries.

Reviewer: Again, this is redundant to the Django documentation, which provides plenty of context and examples.
